Revamp Message Bubbles

itzzritik opened the pull request:
### Changes in this Pull Request
1. Revamp message bubbles
2. Revamp attachments
    - Image attachments
    - URL attachments

Here's Figma for reference ➜ [LINK](https://www.figma.com/file/cenC3oJChduKiAt21DcBml/Friendly-Chat?node-id=436%3A147)

### Screenshots

<img src="https://user-images.githubusercontent.com/24320496/118063985-c7f45e00-b3b7-11eb-91f1-a259b6d658be.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/118063986-c9258b00-b3b7-11eb-9ae3-272b90db54d6.png" width="49%">
<img src="https://user-images.githubusercontent.com/24320496/118063989-c9be2180-b3b7-11eb-839a-0d19910f2b78.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/118063994-ca56b800-b3b7-11eb-921d-dfcfde64e7ad.png" width="49%">

ear-dev committed:
Merge pull request #3 from RocketChat/master

upstream catchup

ear-dev committed:
Merge pull request #5 from RocketChat/master

Master

itzzritik committed:
Revamp Message Bubbles

itzzritik committed:
Merge branch 'master' into feature/revamp-message-bubbles

itzzritik committed:
Fix lints

itzzritik committed:
Add full size emoji support

itzzritik committed:
Fix Lint failure

ear-dev commented:
@ItzzRitik I showed this to our designer and she had a small improvement in relation to how we round corners in the chat bubbles. 

![image](https://user-images.githubusercontent.com/28864717/118299411-b8a40c00-b4ae-11eb-9e86-7e019889e5a1.png)

Can you fix?
thanks!

itzzritik commented:
Eric,
That makes sense, and I was aware of this corner Issue.
Solving this problem was not possible just by CSS (Tried a lot), so I wanted to fix this after developing all the bubbles (message, attachments, location, links etc..) by adding classes from JS itself.

I'll be updating the PR soon.

itzzritik commented:
@ear-dev 
I was able to make the required improvements, Thanks for pointing it out.
I have removed the old **CSS-only** way and Implemented the corners using JS.
Please review the attached screenshots and let me know.

### Screenshots

<img src="https://user-images.githubusercontent.com/24320496/118375660-ba6dec80-b5e0-11eb-8ddb-95e4be3b2632.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/118375659-b9d55600-b5e0-11eb-9f1b-432d92fe2ffc.png" width="49%">

itzzritik committed:
Reimplemented Bubbles Corners

ear-dev commented:
@ItzzRitik this looks good......  I'll also build locally and take a look....  thanks!

ear-dev commented:
@ItzzRitik Hmm... I don't think this is working for me on my local machine.... I'm not seeing the curved corner at the bottom right....   Any ideas?

![image](https://user-images.githubusercontent.com/28864717/118526730-8abd0280-b70e-11eb-9576-af7edcf99112.png)

![image](https://user-images.githubusercontent.com/28864717/118526796-990b1e80-b70e-11eb-93f7-f4a01797cef0.png)

itzzritik commented:
@ear-dev Have you pulled the latest changes?
Want me to check it out?

ear-dev commented:
@ItzzRitik I did pull the latest changes.....  can you test please?  Thanks.

itzzritik committed:
Fix bug on bubble corner for last message

itzzritik commented:
@ear-dev Thanks for pointing, I missed fixing the corner for last message in the conversation.
Now it is fixed, please check

ear-dev commented:
looks good.  thanks.  I would still like for @bhardwajaditya and/or @Shailesh351 to review the code when they have a chance.....  after that we can merge.

ear-dev commented:
@ItzzRitik remind me please.....  are the chat bubble colors configurable?

itzzritik commented:
Hi @ear-dev 
I have updated the code
and Message bubble colours are now configurable

ear-dev commented:
@ItzzRitik did you push the updated code for configurable bubble colors to the branch?  I'm not seeing the commit for that...  Can you also post an image here of where the configuration is done?  Thanks!

itzzritik commented:
@ear-dev 
I have not pushed the changes yet, because it consisted file attachment changes as well.
and I needed some suggestions on that change.

Sure I'll post the screenshots now.

itzzritik commented:
@ear-dev 
In Settings > Layout > Colors > **theme-color-veranda-color-primary**
I'm using already existing **@veranda-color-primary** variable to theme our message bubbles.

## Screenshots  

**Settings**

<img src="https://user-images.githubusercontent.com/24320496/118998873-6cbdff00-b9a7-11eb-8312-85ae540c3ba1.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/118998988-85c6b000-b9a7-11eb-9dd2-2dbaba67bdc8.png" width="49%">

**Chat Room**

<img src="https://user-images.githubusercontent.com/24320496/118998468-1a7cde00-b9a7-11eb-805d-2504fd55cdd4.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/118999174-abec5000-b9a7-11eb-9dcd-ecbd7aa2c9b1.png" width="49%">

ear-dev commented:
@ItzzRitik Another question please........  are you going to add the little check marks that indicate if the message was seen?  Or do you think that should be a separate PR?

ear-dev commented:
@ItzzRitik Also.....  I think we should discuss as a team whether we want chat bubble colors separated from color-primary.  If we did separate it, we would have more options for embedding chat into different containers maybe.

itzzritik commented:
@ear-dev 
We do have read receipts

![image](https://user-images.githubusercontent.com/24320496/119010055-4e5d0100-b9b1-11eb-8c55-b19301410513.png)

ear-dev commented:
@ItzzRitik I'm seeing an issue where the bubbles are not redrawing themselves to conform to the corners until I move away and back to the room.  Thoughts?

![image](https://user-images.githubusercontent.com/28864717/120009608-2588cd00-bfaa-11eb-948b-82498c12a47b.png)

itzzritik committed:
Add separate color settings for message bubbles

itzzritik commented:
@ear-dev 
Sure Eric, I'll look into the issue today

itzzritik commented:
@ear-dev 
## Colour setting for New friendly Chat UI
1. Now we have a colour setting for New friendly chat message bubbles. This setting can be used for other components as well
2. And now file attachments support different icons for different file types:

## Screenshots  

Colour setting for New friendly Chat UI

<img src="https://user-images.githubusercontent.com/24320496/120129084-fdcf7a00-c1e0-11eb-8e6b-099405987f3f.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/120128668-2f941100-c1e0-11eb-9b97-42079f58f1fc.png" width="49%">

File attachment icons

<img src="https://user-images.githubusercontent.com/24320496/120129030-e2646f00-c1e0-11eb-9888-6a0b93788c73.png" width="49%"> <img src="https://user-images.githubusercontent.com/24320496/120128621-18552380-c1e0-11eb-83e9-00fb7ec93b79.png" width="49%">

bhardwajaditya commented on app/ui-message/client/messageBubble.js:
```diff
@@ -0,0 +1,315 @@
+	},
+	own() {
+		const { msg, u = {} } = this;
+		return msg.u && msg.u._id === u._id ? 'own' : 'notOwn';
```
Can we change the classname to make it more clear here

itzzritik replied:
Done, Thanks

bhardwajaditya commented on app/ui-message/client/messageBubble.js:
```diff
@@ -0,0 +1,315 @@
+	},
+});
+
+const hasTempClass = (node) => node.classList.contains('temp');
```
Can we rename it from temp class to actual purpose of this class to make it more clear

itzzritik replied:
This function refers to the actual 'temp class' which was here since before UI migration. and is not something specifically added in this PR.
Probably we shouldn't rename the class definition (as then we may need to change in CSS and settings as well)

I've removed the function itself, Thanks

bhardwajaditya reviewed (commented):
The PR looks good apart from these minor naming conventions.

itzzritik committed:
Fix corner classes

itzzritik commented:
@ear-dev 
On the issue, you mentioned above in the thread,
where bubbles corners were not re-rendering properly.
 
It is now fixed and working as expected.
Please review once again and let me know.
 
Thanks!

ear-dev commented:
@bhardwajaditya I have tested this PR and it looks good to me.  If it looks good to you, hit the 'approve' button, I'll merge it..  thanks!

bhardwajaditya commented:
@ear-dev I have approved it. It looks good now
